plugins/modules/azure_rm_adserviceprincipal_info.py

```diff
@@ -116,10 +122,14 @@ def exec_module(self, **kwargs):
 
         try:
             client = self.get_graphrbac_client(self.tenant)
-            if self.object_id is None:
+            if self.object_id:
+                service_principals = [client.service_principals.get(self.object_id)]
+            elif self.app_id:
                 service_principals = list(client.service_principals.list(filter="servicePrincipalNames/any(c:c eq '{0}')".format(self.app_id)))
+            elif self.app_display_name:
+                service_principals = list(client.service_principals.list(filter="appDisplayName eq '%s'" % self.app_display_name))
```

Comment:
Is the filter condition you have written illegal? Have you tried this locally? Can you share the results? Thank you very much!

```e.g.

invalid query filter clause specified for property 'appDisplayNames'

Reply:
@paulczar kindly ping!

Reply:
I have been using this locally for months, its been working fine.

Reply:
@paulczar `appDisplayName` not working but `displayName` works